src/Analytics/Adapter/ActiveCampaign.php

```diff
@@ -40,27 +41,313 @@ class ActiveCampaign extends Adapter
      */
     private string $actid;
 
+    /**
+     * ActiveCampaign apiKey
+     * @var string
+     */
+    private string $apiKey;
+
     /**
      * Gets the name of the adapter.
      * 
      * @return string
      */
     public function getName(): string
     {
-        return 'ActiveCampaign';
+        return 'Active Campaign';
+    }
+
+
+    /**
+     * Checks if a contact exists and returns a bool if it does.
+     * 
+     * @param string $email
+     * @return bool|int
+     */
+    public function contactExists(string $email): bool|int
+    {
+        try {
+            $result = $this->call('GET', '/api/3/contacts', [], [
+                'email' => $email
+            ]);
+
+            $result = json_decode($result, true);
+
+            if ($result && $result['meta']['total'] > 0) {
+                return $result['contacts'][0]['id'];
+            } else {
+                return false;
+            }
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Create a contact
+     * 
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $phone
+     * @return bool
+     */
+    public function createContact(string $email, string $firstName = '', string $lastName = '', string $phone = ''): bool
+    {
+        $body = ['contact' => [
+            'email' => $email,
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'phone' => $phone
+        ]];
+
+        try {
+            $this->call('POST', '/api/3/contacts', [
+                'Content-Type' => 'application/json'
+            ], $body);
+            return true;
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
```

Comment:
We duplicate this try/catch a lot. Would it be able to move it to Adapter class? Alongside the abstract method, we would have one more method. In there we would do the try/catch wrap. We do something similar in Appwrite regarding workers: https://github.com/appwrite/appwrite/blob/master/src/Appwrite/Resque/Worker.php#L107-L118